src/admin/index.js

```diff
@@ -187,15 +185,11 @@ module.exports = ({
 
     return retrier(async (bail, retryCount, retryTime) => {
       try {
-        await cluster.refreshMetadata()
+        await cluster.refreshMetadata(topics)
         const broker = await cluster.findControllerBroker()
         await broker.deleteTopics({ topics, timeout })
 
-        // Remove deleted topics
-        for (const topic of topics) {
-          cluster.targetTopics.delete(topic)
-        }
-
+        // full metadata refresh to remove recently deleted topics
```

Comment:
Hmm, I don't really like using `refreshMetadata` with an empty array as a way to clear stale metadata. I would rather have an explicit `removeMetadata` or something.

Reply:
Yeah, it isn't so much removing the metadata, as refreshing everything. However, your point is taken and this refresh is unnecessary. I'll add something.